scripts/generate-videos/main.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+
+#  Copyright © 2015-2019 Cask Data, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import argparse
+import os
+import json
+from youtube import API
+
+MAX_RESULTS = 50
+CHANNEL_ID='UCfkRcekMTa5GA2DdNKba7Jg'
+
+api = None
+
+def search_videos(page_token):
+  return api.get('search', part='id,snippet', channelId=CHANNEL_ID, maxResults=MAX_RESULTS, pageToken=page_token, type='video')
+
+def video_preview(video):
+  preview_size = ['maxres', 'high', 'medium', 'standard', 'default']
+  thumbnails = video['snippet']['thumbnails']
+  preview = ''
+  for size in preview_size:
+    if size in thumbnails:
+      preview = thumbnails[size]['url']
+      break
+  return preview
+
+def extract_video_data(video):
+  preview = ''
+  video_data = {}
+  video_data['videoId'] = video['id']['videoId']
+  video_data['title'] = video['snippet']['title']
+  video_data['description'] = video['snippet']['description']
+  video_data['preview'] = video_preview(video)
+  return video_data
+
+def fetch_videos():
+  all_videos = []
+  total_items = MAX_RESULTS
+  page_token = ''
+  while page_token is not None:
+    response = search_videos(page_token)
+    if 'nextPageToken' in response:
+      all_videos = all_videos + list(map(extract_video_data, response['items']))
+      page_token = response['nextPageToken']
+    else:
+      page_token = None
+
+  return all_videos
+
+def get_original_videos(path):
+  try:
+    with open(path) as video_file:
+      return json.load(video_file)
+  except:
+    print('File not found: %s. Will create new one.' % path)
+    return {
+      'videos': []
+    }
+
+def merge_videos(original_videos, youtube_videos, visible, update, update_props):
```

Comment:
what will happen if a video is deleted? It might cause some failures right? We should probably also remove deleted videos

Reply:
done